core/src/test/java/com/google/instrumentation/common/TimestampTest.java

```diff
@@ -36,20 +36,39 @@ public void testTimestampCreateInvalidInput() {
   }
 
   @Test
-  public void testTimestampFromMillis() {
+  public void timestampFromMillis() {
     assertThat(Timestamp.fromMillis(0)).isEqualTo(Timestamp.create(0, 0));
     assertThat(Timestamp.fromMillis(987)).isEqualTo(Timestamp.create(0, 987000000));
     assertThat(Timestamp.fromMillis(3456)).isEqualTo(Timestamp.create(3, 456000000));
   }
 
   @Test
-  public void testTimestampFromMillisNegative() {
+  public void timestampFromMillis_Negative() {
     assertThat(Timestamp.fromMillis(-1)).isEqualTo(Timestamp.create(-1, 999000000));
     assertThat(Timestamp.fromMillis(-999)).isEqualTo(Timestamp.create(-1, 1000000));
     assertThat(Timestamp.fromMillis(-3456)).isEqualTo(Timestamp.create(-4, 544000000));
   }
 
   @Test
+  public void timestampAddNanos() {
+    Timestamp timestamp = Timestamp.create(1234, 223);
+    assertThat(timestamp.addNanos(0)).isEqualTo(timestamp);
+    assertThat(timestamp.addNanos(999999777)).isEqualTo(Timestamp.create(1235, 0));
+    assertThat(timestamp.addNanos(1300200500)).isEqualTo(Timestamp.create(1235, 300200723));
+    assertThat(timestamp.addNanos(1999999777)).isEqualTo(Timestamp.create(1236, 0));
+    assertThat(timestamp.addNanos(9876543789L)).isEqualTo(Timestamp.create(1243, 876544012));
```

Comment:
I think tests that use functions with int/long params should always include cases where max values are used. please add.